test/run-integration-tests.sh

```diff
@@ -116,4 +116,16 @@ else
     echo -e "\nTOTAL FAILURES: $FAILURES OF $COUNT"
 fi
 
+# If we have coverage files, merge them into a single report for upload
+if [ -n "${ARTIFACT_DIRECTORY}" ]; then
+    lcov_args=""
+
+    for info_file in "${ARTIFACT_DIRECTORY}"/*.coverage-info; do
+        lcov_args="${lcov_args} --add-tracefile ${info_file} "
+    done
+    if [ -n "${lcov_args}" ] && [ "${lcov_args}" != "${ARTIFACT_DIRECTORY}/*.coverage-info" ]; then
+        lcov "${lcov_args}" --output-file "${ARTIFACT_DIRECTORY}/merged.coverage-info"
```

Comment:
I think this is not going to work as you might expect, because `"${lcov_args}"` in this case will be considered as a one argument, no matter the spaces, i.e.:

```
$ lcov_args="--xyz --add-tracefile abc"
$ lcov "${lcov_args}"
lcov: Unknown option: xyz --add-tracefile abc
Use lcov --help to get usage information
```

Either you have to unquote `"${lcov_args}"`, which makes the shellcheck sad, or use an array:

```bash
lcov_args=()
...
    lcov_args+=(--add-tracefile "${info_file}")
...
    if [ "${#lcov_args[@]}" -gt 0 ] && ...
        lcov "${lcov_args[@]}" ...
```

Reply:
yes indeed, fixed